src/Service.Tests/Configuration/ConfigurationTests.cs

```diff
@@ -3711,6 +3763,182 @@ public async Task ValidateNextLinkUsage()
             }
         }
 
+        /// <summary>
+        /// Tests the enforcement of depth limit restrictions on GraphQL queries and mutations in non-hosted mode.
+        /// Verifies that requests exceeding the specified depth limit result in a BadRequest, 
+        /// while requests within the limit succeed with the expected status code.
+        /// Also verifies that the error message contains the current and allowed max depth limit value.
+        /// </summary>
+        /// <param name="depthLimit">The maximum allowed depth for GraphQL queries and mutations.</param>
+        /// <param name="isMutationOperation">Indicates whether the operation is a mutation (true) or a query (false).</param>
+        /// <param name="expectedStatusCodeForGraphQL">The expected HTTP status code for the operation.</param>
+        [DataTestMethod]
+        [DataRow(1, false, HttpStatusCode.BadRequest, DisplayName = "Failed Query execution when max depth limit is set to 1")]
+        [DataRow(2, false, HttpStatusCode.OK, DisplayName = "Query execution succesful when max depth limit is set to 2")]
+        [DataRow(1, true, HttpStatusCode.BadRequest, DisplayName = "Failed Mutation execution when max depth limit is set to 1")]
+        [DataRow(2, true, HttpStatusCode.OK, DisplayName = "Mutation execution succesful when max depth limit is set to 2")]
+        [TestCategory(TestCategory.MSSQL)]
+        public async Task TestDepthLimitRestictionOnGrahQLInNonHostedMode(
+            int depthLimit,
+            bool isMutationOperation,
+            HttpStatusCode expectedStatusCodeForGraphQL)
+        {
+            // Arrange
+            GraphQLRuntimeOptions graphqlOptions = new(DepthLimit: depthLimit);
+            graphqlOptions = graphqlOptions with { UserProvidedDepthLimit = true };
+
+            DataSource dataSource = new(DatabaseType.MSSQL,
+                GetConnectionStringFromEnvironmentConfig(environment: TestCategory.MSSQL), Options: null);
+
+            RuntimeConfig configuration = InitMinimalRuntimeConfig(dataSource, graphqlOptions, restOptions: new());
+            const string CUSTOM_CONFIG = "custom-config.json";
+            File.WriteAllText(CUSTOM_CONFIG, configuration.ToJson());
+
+            string[] args = new[]
+            {
+                $"--ConfigFileName={CUSTOM_CONFIG}"
+            };
+
+            // Act
+            using (TestServer server = new(Program.CreateWebHostBuilder(args)))
+            using (HttpClient client = server.CreateClient())
+            {
+                string query;
+                if (isMutationOperation)
+                {
+                    // requested mutation operation has depth of 2
+                    query = GetSimpleGraphQLCreateMutation();
```

Comment:
```suggestion
                    request = GetSimpleGraphQLCreateMutation();
```

query != mutation, can use more generic term: GraphQL Request

Reply:
for a graphql to process the request, it should be in the following format 
{ query = "{ book_by_pk(id: 1) {
                       id,
                       title,
                      publisher_id
                   }
}" }

{ query = "mutation createbook{
     createbook(item: { title: \"Book #1\", publisher_id: 1234 }) {\n                            
         title
         publisher_id
     }
 }" }
 
 that's why it's named specifically query

Reply:
So GraphQL endpoint expects request body to have top level property query? 
To answer my question and elaborate:

The GraphQL request over http: https://graphql.org/learn/serving-over-http/
results in populating the query param `query` or having an `application/json` type request body with contents: 
```json
{
    "query": "query book_by_pk{\r\nbook_by_pk(id: 1) {\r\nid,\r\ntitle,\r\n publisher_id\r\n  }\r\n  }"
}
```